### PR TITLE
Added tests for pygame.transform.smoothscale(); fixes #1758

### LINKED ISSUE #1758: Add unit test: transform.smoothscale()
This function needs a unit test.

[Guide to how to contribute to an open source project on GitHub.](https://github.com/firstcontributions/first-contributions)

**Test stub:** https://github.com/pygame/pygame/blob/master/test/transform_test.py#L1043-L1062
**Docs:** https://www.pygame.org/docs/ref/transform.html#pygame.transform.smoothscale
**Code to be tested:** https://github.com/pygame/pygame/blob/master/src_c/transform.c#L1416-L1488

### PR DESCRIPTION
**Pull request by adamandrews1**

### CODE REVIEW

**Comment by adamandrews1** [comment]
Fixes #1758 . We may want to add a more sophisticated pixel-color-blending test portion. @MyreMylar

File: test/transform_test.py [hunk 1]
```diff
@@ -1176,8 +1176,33 @@ def todo_test_smoothscale(self):
...
+            smoothscaled_surface = pygame.transform.smoothscale(starting_surface, (10, 10))
+        self.assertRaises(ValueError, smoothscale_low_bpp)
+
+        def smoothscale_high_bpp():
+            starting_surface = pygame.Surface((20, 20), depth=48)
+            smoothscaled_surface = pygame.transform.smoothscale(starting_surface, (10, 10))
+        self.assertRaises(ValueError, smoothscale_high_bpp)
+
+        def smoothscale_invalid_scale():
+            starting_surface = pygame.Surface((20, 20), depth=36)
```
Comment (MyreMylar) [inline test/transform_test.py:1191]: should the depth here be 32 ?

**Review by MyreMylar — CHANGES_REQUESTED** [review]
I suspect you could write a reasonable test of the blending filter algorithm by doing something like:

```
two_pixel_surface = pygame.Surface((2, 1), depth=32)
two_pixel_surface.fill(pygame.Color(0, 0, 0), pygame.Rect(0, 0, 1, 1))
two_pixel_surface.fill(pygame.Color(255, 255, 255), pygame.Rect(1, 0, 1, 1))

two_five_six_surf = pygame.transform.smoothscale(two_pixel_surface, (256, 1))
self.assertEqual(two_five_six_surf.get_at((128, 0)), pygame.Color(127, 127, 127))
```

And perhaps something similar starting larger and going the other way to see if you get roughly a midway blend at the halfway point.

File: test/transform_test.py (hunk 1 above)
Comment (adamandrews1) [reply test/transform_test.py:1191]: It definitely should be. This is not the value error we are looking for in this test! Thanks

**Review by MyreMylar — APPROVED** [review]
Looks good, there is another (already existing) test that covers some other functionality of smoothscale and some tests of setting it's backend as well so I'm happy that we have pretty good coverage of it with this test as well.

Thanks for the contribution! 👍 🎉

**Comment by adamandrews1** [comment]
Thanks for the thoughtful guidance!

**Pull request merged**